util/src/main/java/io/kubernetes/client/util/ConfigBuilder.java

```diff
@@ -0,0 +1,285 @@
+
+	public ConfigBuilder setCertificateAuthorityFile(String certificateAuthorityFile) {
+		this.certificateAuthorityFile = certificateAuthorityFile;
+		return this;
+
+	}
+
+	public String getCertificateAuthorityData() {
+		return certificateAuthorityData;
+	}
+
+	public ConfigBuilder setCertificateAuthorityData(String certificateAuthorityData) {
+		this.certificateAuthorityData = certificateAuthorityData;
+		return this;
+	}
+
+	public ConfigBuilder setClusterMode(boolean clusterMode) {
+		this.clusterMode = clusterMode;
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(KubeConfig config) {
+		this.kubeConfig = config;
+		return this;
+	}
+
+	public ConfigBuilder setDefaultKubeConfigMode(boolean defaultKubeConfigMode) {
+		this.defaultKubeConfigMode = defaultKubeConfigMode;
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(String fileName) throws FileNotFoundException {
+		this.kubeConfig = KubeConfig.loadKubeConfig(new FileReader(fileName));
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(Reader input)  {
+		this.kubeConfig = KubeConfig.loadKubeConfig(input);
+		return this;
+	}
+
+	public ConfigBuilder setKubeConfig(InputStream stream)  {
+		this.kubeConfig = KubeConfig.loadKubeConfig(new InputStreamReader(stream));
+		return this;
+	}
+
+	public KeyManager[] getKeyMgrs() {
+		return keyMgrs;
+	}
+
+	public ConfigBuilder setKeyMgrs(KeyManager[] keyMgrs) {
+		this.keyMgrs = keyMgrs;
+		return this;
+	}
+
+	public boolean isVerifyingSsl() {
+		return verifyingSsl;
+	}
+
+	public ConfigBuilder setVerifyingSsl(boolean verifyingSsl) {
+		this.verifyingSsl = verifyingSsl;
+		return this;
+	}
+
+	public boolean isDefaultClientMode() {
+		return defaultClientMode;
+	}
+
+	public ConfigBuilder setDefaultClientMode(boolean defaultClientMode) {
+		this.defaultClientMode = defaultClientMode;
+		return this;
+	}
+
+	public String getApiKeyPrefix() {
+		return apiKeyPrefix;
+	}
+
+	public ConfigBuilder setApiKeyPrefix(String apiKeyPrefix) {
+		this.apiKeyPrefix = apiKeyPrefix;
+		return this;
+	}
+
+	public ApiClient build()  {
+
+		ApiClient client = new ApiClient();
+
+		if( kubeConfig !=null) {
+			client = Config.fromConfig(kubeConfig);
+			return client;
+		}
+
+		if(defaultKubeConfigMode == true) {
+			try {
+				client = Config.fromConfig(KubeConfig.loadDefaultKubeConfig());
+			} catch (FileNotFoundException e) {
+				e.printStackTrace();
+			}
+			return client;
+		}
+
+		if(clusterMode == true) {
+			try {
+				client = Config.fromCluster();
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+			return client;
+		}
+
+		if(defaultClientMode ==true ) {
+			try {
+				client = Config.defaultClient();
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+			return client;
+
+		}
+
+		if (basePath != null ) {
+			if(basePath.endsWith("/")) {
+				basePath = basePath.substring(0, basePath.length() - 1);
+			}
+			client.setBasePath(basePath)
+			.setVerifyingSsl(verifyingSsl);
+		}
+
+		else {
+			try {
```

Comment:
This is weird. You are `try{ ... } catch` and then throwing the exception yourself.  Just log to stderr? or throw an exception?